sql/000_001_006-name_cleaning.sql

```diff
@@ -0,0 +1,130 @@
+/* this file will create a table defining substituations for simplifying */
+/* company names in preparation for fuzzy matching */ 
+
+CREATE TABLE etl.company_name_abbreviations (
+  to_replace TEXT NOT NULL UNIQUE,
+  replacement TEXT NOT NULL,
+  regexp_flags TEXT NOT NULL DEFAULT 'ig'
+);
+CREATE INDEX ix_company_name_replacement_cover ON etl.company_name_abbreviations (
+  to_replace,
+  replacement,
+  regexp_flags
+);
+
+INSERT INTO etl.company_name_abbreviations (
+  to_replace,
+  replacement,
+  regexp_flags
+) VALUES
+('\s+and\s+', ' & ', 'ig'),
+('\s+en\s+', ' & ', 'ig'),
+('\s+och\s+', ' & ', 'ig'),
+('\s+und\s+', ' & ', 'ig'),
+('\(inactive\)', '', 'ig'),
+('aktg', 'ag', 'ig'),
+('aktiengesellschaft', 'ag', 'ig'),
+/* covers associate AND associates, single and plural */
+('associate[s]*', 'assoc', 'ig'),
+('berhad', 'bhd', 'ig'),
+('company', 'co', 'ig'),
+('corporation', 'corp', 'ig'),
+('designated\s+activity[\s\$]+co', 'dac', 'ig'),
+('develop(ment){0,1}', 'dev', 'ig'),
+('financial', 'fin', 'ig'),
+('generation', 'gen', 'ig'),
+('government', 'govt', 'ig'),
+('group', 'grp', 'ig'),
+('holding(s){0,1}', 'hldgs', 'ig'),
+('incorporated', 'inc', 'ig'),
+('international', 'intl', 'ig'),
+('investment', 'invest', 'ig'),
+/* lp: limited -> ltd AND partnership -> prthip, so this turns variants into lp */
+('ltd\s+prthip', 'lp', 'ig'),
```

Comment:
Will this catch cases where there is no 'limited' or 'ltd' in the string, ie if partnership -> prthip will the final result include prthip? I think we should try to get partnership -> prt in that case

Reply:
If there's no limited or ltd, then it shouldn't be collapsing to `lp` But yes, `partnership` collpases to `prthip` in the current model.